libraries/AP_BattMonitor/AP_BattMonitor_Sum.cpp

```diff
@@ -0,0 +1,56 @@
+#include <AP_HAL/AP_HAL.h>
+#include <AP_Common/AP_Common.h>
+#include <AP_Math/AP_Math.h>
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_Sum.h"
+
+/*
+  battery monitor that is the sum of other battery monitors.
+
+  This can be used to combined other current/voltage sensors into a
+  single backend
+ */
+extern const AP_HAL::HAL& hal;
+
+/// Constructor
+AP_BattMonitor_Sum::AP_BattMonitor_Sum(AP_BattMonitor &mon,
+                                       AP_BattMonitor::BattMonitor_State &mon_state,
+                                       AP_BattMonitor_Params &params) :
+    AP_BattMonitor_Backend(mon, mon_state, params)
+{
+}
+
+// read - read the voltage and current
+void
+AP_BattMonitor_Sum::read()
+{
+    float voltage_sum = 0;
+    uint8_t voltage_count = 0;
+    float current_sum = 0;
+    uint8_t current_count = 0;
+
+    for (uint8_t i=0; i<_mon.num_instances(); i++) {
```

Comment:
Apparently I never posted the main comment I was trying to make here. (Sorry about that).

The discussion with @OXINARF indicated that this was to sum all instances after the selected sum index. As this is it sums all the instances. If we start from `get_instance() + 1` then you don't have to have the early continue and you can still mix/match other monitors earlier in the system safely.

Reply:
done